File: cpp/lexer/word.cpp
#include <lexer/word.hpp>

namespace lexer {
std::shared_ptr<Word> Word::and_word = Word::create("&&", Token::kAnd);
std::shared_ptr<Word> Word::or_word = Word::create("||", Token::kOr);
std::shared_ptr<Word> Word::equal = Word::create("==", Token::kEqual);
std::shared_ptr<Word> Word::not_equal = Word::create("!=", Token::kNotEqual);
std::shared_ptr<Word> Word::less_equal = Word::create("<=", Token::kLessEqual);
std::shared_ptr<Word> Word::greater_equal = Word::create(">=", Token::kGreaterEqual);
std::shared_ptr<Word> Word::minus = Word::create("minus", Token::kMinus);
std::shared_ptr<Word> Word::true_word = Word::create("true", Token::kTrue);
std::shared_ptr<Word> Word::false_word = Word::create("false", Token::kFalse);
std::shared_ptr<Word> Word::temp = Word::create("t", Token::kTemp);
} // namespace lexer

File: java/com/dragon/lexer/Word.java
package com.dragon.lexer;

public class Word extends Token {
  public static final Word and = new Word("&&", Tag.AND);
  public static final Word or = new Word("||", Tag.OR);
  public static final Word eq = new Word("==", Tag.EQ);
  public static final Word ne = new Word("!=", Tag.NE);
  public static final Word le = new Word("<=", Tag.LE);
  public static final Word ge = new Word(">=", Tag.GE);
  public static final Word minus = new Word("minus", Tag.MINUS);
  public static final Word True = new Word("true", Tag.TRUE);
  public static final Word False = new Word("false", Tag.FALSE);
  public static final Word temp = new Word("t", Tag.TEMP);

  private String lexeme = "";

  public Word(String s, int tag) {
    super(tag);
    lexeme = s;
  }

  public String getLexeme() {
    return lexeme;
  }

  public String toString() {
    return lexeme;
  }
}
